devdoc/DevTestWithLDAP.md

```diff
@@ -0,0 +1,146 @@
+# Overview
+
+These directions walk through the steps to configure and setup, from scratch, an out-of-the-box HPCC platform deployment using LDAP authentication.
+
+The goal is for developers and other technical platform users to have available an LDAP Directory Server (DS) server they can use for testing and troubleshooting authentication and authorization features. These directions don't cover the steps needed for securing a production-grade DS instance and the communication between it and the platform components. Any deployment with this setup should be ephemeral, publicly unreachable, and not have access to sensitive data.
+
+First we will set up a Docker container with a minimal 389ds LDAP Directory Server and a phpLdapAdmin (PLA) client to administer it. The PLA can be used to inspect your DS and perform any administrative tasks not supported by the ECL Watch client.
+
+Then we'll cover the configuration changes to deploy a minimal HPCC platform, both non-containerized (bare-metal) and under Kubernetes, that uses 389ds for security.
+
+## Initial State
+
+When the platform launches it ensures the LDAP DS is initialized with "OUs" (organizational units) needed. OUs for groups, users and esp authorization resources are created (among others). An HPCC Admin user is created and added to the administrator's group.
+
+If you need any other security settings beyond that, such as non-administrative users and groups, you'll need to create them using ECL Watch. The administrator account itself has the privileges to make these kinds of changes. You may see that the admin doesn't by default have _all_ permissions, but it can add any it needs to the admin group.
+
+This setup has been tested with ECL Watch authentication and authorization. It should also work for file and workunit scopes, but that hasn't been confirmed.
+
+Your customized setup will be saved to a persistent volume, and you can separately maintain different versions as needed.
+
+# Docker LDAP Directory Service and Admin Setup
+
+We'll run our 389ds and PLA in a Docker container launched from platform source: `dockerfiles/examples/ldap`.
+
+These instructions create a persistent volume for the DS inside your directory, but you can customize the docker-compose file to use another location. All of the server state is stored in the mounted volume, so if you want to be able to easily switch between different configurations you can easily do so by switching the mounted directory at launch.
+
+1. Edit the `dockerfiles/examples/ldap/docker-compose.yaml` file, customizing the admin password and PVC mount point:
```

Comment:
I would recommend that you have them copy these config files elsewhere so that their repo is not in a modified state.

Reply:
Ok adjusted instructions.